src/core/utils/urlutils.js

```diff
@@ -82,3 +84,62 @@ export function equivalentParams (params1, params2) {
   }
   return true;
 }
+
+/**
+ * Creates a copy of the provided {@link SearchParams}, with the specified
+ * attributes filtered out
+ * @param {SearchParams} params The parameters to remove from
+ * @param {string[]} prefixes The prefixes of parameters to remove
+ * @return {SearchParams} A new instance of SearchParams without removed params
+ *   from the params parameter
+ */
+export function removeParamsWithPrefixes (params, prefixes) {
+  const newParams = new SearchParams();
+  for (const [key, val] of params.entries()) {
+    const includeEntry = prefixes.every(prefix => !key.startsWith(prefix));
+    if (includeEntry) {
+      newParams.set(key, val);
+    }
+  }
+  return newParams;
+}
+
+/**
+ * Removes parameters for filters, facets, sort options, and pagination
+ * from the provided {@link SearchParams}. This is useful for constructing
+ * inter-experience answers links.
+ * @param {SearchParams} params The parameters to remove from
+ * @param {function} getComponentNamesForComponentTypes Given string[]
+ *   component types, returns string[] component names for those types
+ * @return {SearchParams} Parameters that have filtered out params that
+ *   should not persist across the answers experience
+ */
+export function filterParamsForExperienceLink (
+  params,
+  getComponentNamesForComponentTypes
+) {
+  const prefixComponentTypes = [
+    ComponentTypes.FACETS,
+    ComponentTypes.FILTER_BOX,
+    ComponentTypes.FILTER_OPTIONS,
+    ComponentTypes.RANGE_FILTER,
+    ComponentTypes.DATE_RANGE_FILTER,
+    ComponentTypes.SORT_OPTIONS
+  ];
+  const queryPrefixComponentComponentTypes = [
+    ComponentTypes.GEOLOCATION_FILTER,
+    ComponentTypes.FILTER_SEARCH
+  ];
+  let prefixes = [
+    StorageKeys.FILTER
+  ];
+  prefixes = prefixes.concat(getComponentNamesForComponentTypes(prefixComponentTypes));
+  prefixes = prefixes.concat(
+    getComponentNamesForComponentTypes(queryPrefixComponentComponentTypes)
+      .map((name) => { return `${StorageKeys.QUERY}.${name}`; })
+  );
+
+  const newParams = removeParamsWithPrefixes(params, prefixes);
+  newParams.delete(StorageKeys.SEARCH_OFFSET);
+  return newParams;
```

Comment:
I think we should try and streamline this method a bit. It is a bit difficult to read as is. One thing that would help readability is potentially some different variable names.

prefixComponentTypes -> componentTypesToExclude

componentTypesToExclude = [
    ComponentTypes.FACETS,
    ComponentTypes.FILTER_BOX,
    ComponentTypes.FILTER_OPTIONS,
    ComponentTypes.RANGE_FILTER,
    ComponentTypes.DATE_RANGE_FILTER,
    ComponentTypes.SORT_OPTIONS,
    ComponentTypes.GEOLOCATION_FILTER,
    ComponentTypes.FILTER_SEARCH
  ];

let paramsToFilter = componentTypesToExclude.map(type => {
    let typeFilter = getComponentNamesForComponentTypes(type)
    if (type === ComponentTypes.GEOLOCATION_FILTER || type === ComponentTypes.FILTER_SEARCH) {
       typeFilter === `${StorageKeys.QUERY}.${typeFilter}`;
    }
   return typeFilter;
});
paramsToFilter.push(StorageKeys.FILTER);

I think the above might be a bit more readable. There's only one concat and map and we don't have two partial lists that need to be run through getComponentNamesForComponentTypes, then re-combined.

Reply:
So getComponentNamesForComponentTypes returns an array, as there can be many names, so we'd need to reduce otherwise we'd have an array of arrays.

I'm not sure how much this lends itself to being readable. What do you think?

```
  const componentTypesToExclude = [
    ComponentTypes.FACETS,
    ComponentTypes.FILTER_BOX,
    ComponentTypes.FILTER_OPTIONS,
    ComponentTypes.RANGE_FILTER,
    ComponentTypes.DATE_RANGE_FILTER,
    ComponentTypes.SORT_OPTIONS
    ComponentTypes.GEOLOCATION_FILTER,
    ComponentTypes.FILTER_SEARCH
  ];
  let prefixes = [
    StorageKeys.FILTER
  ];
  prefixes = prefixes.concat(
    componentTypesToExclude
    .reduce((currentPrefixes, type) => {
      const prefixesToAdd = getComponentNamesForComponentTypes([type]);
      if (type === ComponentTypes.GEOLOCATION_FILTER || type === ComponentTypes.FILTER_SEARCH) {
        prefixesToAdd = prefixesToAdd.map(p => `${StorageKeys.QUERY}.${p}`);
      }
      return currentPrefixes.concat();
    }, []);
  );
```

Reply:
Passed this along on Slack as well, but what about:
```
  const componentTypesToExclude = [
    ComponentTypes.FACETS,
    ComponentTypes.FILTER_BOX,
    ComponentTypes.FILTER_OPTIONS,
    ComponentTypes.RANGE_FILTER,
    ComponentTypes.DATE_RANGE_FILTER,
    ComponentTypes.SORT_OPTIONS
    ComponentTypes.GEOLOCATION_FILTER,
    ComponentTypes.FILTER_SEARCH
  ];
  let paramsToFilter = componentTypesToExclude.flatMap(type => {
     let params = getComponentNamesForComponentTypes(type);
      if (type === ComponentTypes.GEOLOCATION_FILTER || type === ComponentTypes.FILTER_SEARCH) {
        params = params.map(p => `${StorageKeys.QUERY}.${p}`);
      }
      return params;
  }
  paramsToFilter.concat([StorageKeys.FILTER]);
```